# NBA All-Star Weekend Challenge

Your task...answer the following questions given the dataset given about the NBA All-Star Game from the year 2000 - 2016

Become more familiar with the dataset <a href="https://www.kaggle.com/fmejia21/nba-all-star-game-20002016">here</a>

Once completed, commit this notebook to github and submit the link to the google classroom assignment.

<a href="https://classroom.google.com/u/2/c/NDc4MzEzMjI5Nzla/a/NTE3OTYxNzM2OTNa/details">Google Classroom</a>

In [2]:
import pandas as pd
import numpy as np

In [57]:
df = pd.read_csv("NBA All Stars 2000-2016 - Sheet1.csv")

In [32]:
df.head()

Year         Player Pos    HT   WT                   Team  \
0  2016  Stephen Curry   G   6-3  190  Golden State Warriors   
1  2016   James Harden  SG   6-5  220        Houston Rockets   
2  2016   Kevin Durant  SF   6-9  240  Golden State Warriors   
3  2016  Kawhi Leonard   F   6-7  230      San Antonio Spurs   
4  2016  Anthony Davis  PF  6-11  253   New Orleans Pelicans   

                        Selection Type    NBA Draft Status    Nationality  
0  Western All-Star Fan Vote Selection   2009 Rnd 1 Pick 7  United States  
1  Western All-Star Fan Vote Selection   2009 Rnd 1 Pick 3  United States  
2  Western All-Star Fan Vote Selection   2007 Rnd 1 Pick 2  United States  
3  Western All-Star Fan Vote Selection  2011 Rnd 1 Pick 15  United States  
4  Western All-Star Fan Vote Selection   2012 Rnd 1 Pick 1  United States

In [33]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 439 entries, 0 to 438
Data columns (total 9 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   Year              439 non-null    int64 
 1   Player            439 non-null    object
 2   Pos               439 non-null    object
 3   HT                439 non-null    object
 4   WT                439 non-null    int64 
 5   Team              439 non-null    object
 6   Selection Type    439 non-null    object
 7   NBA Draft Status  439 non-null    object
 8   Nationality       439 non-null    object
dtypes: int64(2), object(7)
memory usage: 31.0+ KB


# What is the average weight of all players who played during this time?

In [10]:
avg_weight = df.WT.mean()

In [11]:
avg_weight

228.75170842824602

#### Average weight was 228.75

# What team is represented the most during this time?

In [12]:
most_rep = df.Team.mode()

In [13]:
most_rep

0    Miami Heat
dtype: object

#### Miami heat most represented 

# What draft class is represeted the most during this time?

In [58]:
df["Draft_Class"] = df["NBA Draft Status"].str.split(" ", expand = False) 

In [59]:
df.head()

Year         Player Pos    HT   WT                   Team  \
0  2016  Stephen Curry   G   6-3  190  Golden State Warriors   
1  2016   James Harden  SG   6-5  220        Houston Rockets   
2  2016   Kevin Durant  SF   6-9  240  Golden State Warriors   
3  2016  Kawhi Leonard   F   6-7  230      San Antonio Spurs   
4  2016  Anthony Davis  PF  6-11  253   New Orleans Pelicans   

                        Selection Type    NBA Draft Status    Nationality  \
0  Western All-Star Fan Vote Selection   2009 Rnd 1 Pick 7  United States   
1  Western All-Star Fan Vote Selection   2009 Rnd 1 Pick 3  United States   
2  Western All-Star Fan Vote Selection   2007 Rnd 1 Pick 2  United States   
3  Western All-Star Fan Vote Selection  2011 Rnd 1 Pick 15  United States   
4  Western All-Star Fan Vote Selection   2012 Rnd 1 Pick 1  United States   

                Draft_Class  
0   [2009, Rnd, 1, Pick, 7]  
1   [2009, Rnd, 1, Pick, 3]  
2   [2007, Rnd, 1, Pick, 2]  
3  [2011, Rnd, 1, Pick, 15]  
4   [2012, Rnd, 1, Pick, 1]

In [60]:
df[["drft_yr", "rnd", "rnd_num","pck", "pck_num"]] = pd.DataFrame(df.Draft_Class.tolist(), index= df.index)

In [61]:
df.head()

Year         Player Pos    HT   WT                   Team  \
0  2016  Stephen Curry   G   6-3  190  Golden State Warriors   
1  2016   James Harden  SG   6-5  220        Houston Rockets   
2  2016   Kevin Durant  SF   6-9  240  Golden State Warriors   
3  2016  Kawhi Leonard   F   6-7  230      San Antonio Spurs   
4  2016  Anthony Davis  PF  6-11  253   New Orleans Pelicans   

                        Selection Type    NBA Draft Status    Nationality  \
0  Western All-Star Fan Vote Selection   2009 Rnd 1 Pick 7  United States   
1  Western All-Star Fan Vote Selection   2009 Rnd 1 Pick 3  United States   
2  Western All-Star Fan Vote Selection   2007 Rnd 1 Pick 2  United States   
3  Western All-Star Fan Vote Selection  2011 Rnd 1 Pick 15  United States   
4  Western All-Star Fan Vote Selection   2012 Rnd 1 Pick 1  United States   

                Draft_Class drft_yr  rnd rnd_num   pck pck_num  
0   [2009, Rnd, 1, Pick, 7]    2009  Rnd       1  Pick       7  
1   [2009, Rnd, 1, Pick, 3]    2009  Rnd       1  Pick       3  
2   [2007, Rnd, 1, Pick, 2]    2007  Rnd       1  Pick       2  
3  [2011, Rnd, 1, Pick, 15]    2011  Rnd       1  Pick      15  
4   [2012, Rnd, 1, Pick, 1]    2012  Rnd       1  Pick       1

In [62]:
df.drop(['Draft_Class','rnd', 'pck'], axis=1)

Year           Player Pos    HT   WT                    Team  \
0    2016    Stephen Curry   G   6-3  190   Golden State Warriors   
1    2016     James Harden  SG   6-5  220         Houston Rockets   
2    2016     Kevin Durant  SF   6-9  240   Golden State Warriors   
3    2016    Kawhi Leonard   F   6-7  230       San Antonio Spurs   
4    2016    Anthony Davis  PF  6-11  253    New Orleans Pelicans   
..    ...              ...  ..   ...  ...                     ...   
434  2000  Antonio McDyess   F   6-9  245          Denver Nuggets   
435  2000      Gary Payton   G   6-4  180     Seattle SuperSonics   
436  2000  Rasheed Wallace  FC  6-11  230  Portland Trail Blazers   
437  2000   David Robinson   C   7-1  250       San Antonio Spurs   
438  2000      Vlade Divac   C   7-1  260        Sacramento Kings   

                             Selection Type    NBA Draft Status  \
0       Western All-Star Fan Vote Selection   2009 Rnd 1 Pick 7   
1       Western All-Star Fan Vote Selection   2009 Rnd 1 Pick 3   
2       Western All-Star Fan Vote Selection   2007 Rnd 1 Pick 2   
3       Western All-Star Fan Vote Selection  2011 Rnd 1 Pick 15   
4       Western All-Star Fan Vote Selection   2012 Rnd 1 Pick 1   
..                                      ...                 ...   
434      Western All-Star Coaches Selection   1995 Rnd 1 Pick 2   
435      Western All-Star Coaches Selection   1990 Rnd 1 Pick 2   
436      Western All-Star Coaches Selection   1995 Rnd 1 Pick 4   
437      Western All-Star Coaches Selection   1987 Rnd 1 Pick 1   
438  Western All-Star Replacement Selection  1989 Rnd 1 Pick 26   

       Nationality drft_yr rnd_num pck_num  
0    United States    2009       1       7  
1    United States    2009       1       3  
2    United States    2007       1       2  
3    United States    2011       1      15  
4    United States    2012       1       1  
..             ...     ...     ...     ...  
434  United States    1995       1       2  
435  United States    1990       1       2  
436  United States    1995       1       4  
437  United States    1987       1       1  
438         Serbia    1989       1      26  

[439 rows x 12 columns]

In [63]:
draft_class_rep = df["drft_yr"].mode()

In [64]:
draft_class_rep

0    1996
dtype: object

#### 1996 is represented the most 

# Are foriegn players more prevalent in the All-Star Game during this time?

In [65]:
df["foreign_or_not"] = np.where(df["Nationality"]=="United States", "US", "No")

In [66]:
df

Year           Player Pos    HT   WT                    Team  \
0    2016    Stephen Curry   G   6-3  190   Golden State Warriors   
1    2016     James Harden  SG   6-5  220         Houston Rockets   
2    2016     Kevin Durant  SF   6-9  240   Golden State Warriors   
3    2016    Kawhi Leonard   F   6-7  230       San Antonio Spurs   
4    2016    Anthony Davis  PF  6-11  253    New Orleans Pelicans   
..    ...              ...  ..   ...  ...                     ...   
434  2000  Antonio McDyess   F   6-9  245          Denver Nuggets   
435  2000      Gary Payton   G   6-4  180     Seattle SuperSonics   
436  2000  Rasheed Wallace  FC  6-11  230  Portland Trail Blazers   
437  2000   David Robinson   C   7-1  250       San Antonio Spurs   
438  2000      Vlade Divac   C   7-1  260        Sacramento Kings   

                             Selection Type    NBA Draft Status  \
0       Western All-Star Fan Vote Selection   2009 Rnd 1 Pick 7   
1       Western All-Star Fan Vote Selection   2009 Rnd 1 Pick 3   
2       Western All-Star Fan Vote Selection   2007 Rnd 1 Pick 2   
3       Western All-Star Fan Vote Selection  2011 Rnd 1 Pick 15   
4       Western All-Star Fan Vote Selection   2012 Rnd 1 Pick 1   
..                                      ...                 ...   
434      Western All-Star Coaches Selection   1995 Rnd 1 Pick 2   
435      Western All-Star Coaches Selection   1990 Rnd 1 Pick 2   
436      Western All-Star Coaches Selection   1995 Rnd 1 Pick 4   
437      Western All-Star Coaches Selection   1987 Rnd 1 Pick 1   
438  Western All-Star Replacement Selection  1989 Rnd 1 Pick 26   

       Nationality               Draft_Class drft_yr  rnd rnd_num   pck  \
0    United States   [2009, Rnd, 1, Pick, 7]    2009  Rnd       1  Pick   
1    United States   [2009, Rnd, 1, Pick, 3]    2009  Rnd       1  Pick   
2    United States   [2007, Rnd, 1, Pick, 2]    2007  Rnd       1  Pick   
3    United States  [2011, Rnd, 1, Pick, 15]    2011  Rnd       1  Pick   
4    United States   [2012, Rnd, 1, Pick, 1]    2012  Rnd       1  Pick   
..             ...                       ...     ...  ...     ...   ...   
434  United States   [1995, Rnd, 1, Pick, 2]    1995  Rnd       1  Pick   
435  United States   [1990, Rnd, 1, Pick, 2]    1990  Rnd       1  Pick   
436  United States   [1995, Rnd, 1, Pick, 4]    1995  Rnd       1  Pick   
437  United States   [1987, Rnd, 1, Pick, 1]    1987  Rnd       1  Pick   
438         Serbia  [1989, Rnd, 1, Pick, 26]    1989  Rnd       1  Pick   

    pck_num foreign_or_not  
0         7             US  
1         3             US  
2         2             US  
3        15             US  
4         1             US  
..      ...            ...  
434       2             US  
435       2             US  
436       4             US  
437       1             US  
438      26             No  

[439 rows x 16 columns]

In [79]:
foreign = df['foreign_or_not'].value_counts()
foreign

US    365
No     74
Name: foreign_or_not, dtype: int64

#### Most players are from the US (365) vs foreign (74)

# How often are the Western All-Stars voted in by fans?

In [82]:
west_fans = df['Selection Type'].value_counts()
west_fans 

Eastern All-Star Coaches Selection        119
Western All-Star Coaches Selection        119
Western All-Star Fan Vote Selection        85
Eastern All-Star Fan Vote Selection        85
Eastern All-Star Replacement Selection     16
Western All-Star Replacement Selection     15
Name: Selection Type, dtype: int64

#### The Western All-Stars are voted in by fans 85 times 

# How many times does Steph Curry make the All-Star game during this time period?

In [92]:
steph_freq = df[(df.Player=="Stephen Curry")].count()
steph_freq 

Year                4
Player              4
Pos                 4
HT                  4
WT                  4
Team                4
Selection Type      4
NBA Draft Status    4
Nationality         4
Draft_Class         4
drft_yr             4
rnd                 4
rnd_num             4
pck                 4
pck_num             4
foreign_or_not      4
dtype: int64

#### Steph makes it 4 times to the all star game during this time 

# How many Shooting Gaurds have made the All-Star game during this time?

In [121]:
sg = df[(df.Pos=="SG")]  
sg_unique = sg.Player.nunique() 
sg_unique


9

#### 9 Shooting guards made the all star team during this time 

# How many Chicago Bulls players have made the All-Star Game during this time period?

In [124]:
bulls = df[(df.Team=="Chicago Bulls")]
bulls_unique = bulls.Player.nunique()
bulls_unique

5

#### 5 Chicago Bulls players have made the All-Star Game during this time period

In [125]:
bulls

Year        Player Pos    HT   WT           Team  \
16   2016  Jimmy Butler  GF   6-7  231  Chicago Bulls   
29   2015  Jimmy Butler  GF   6-7  231  Chicago Bulls   
36   2015     Pau Gasol  FC   7-0  250  Chicago Bulls   
53   2014     Pau Gasol  FC   7-0  250  Chicago Bulls   
58   2014  Jimmy Butler  GF   6-7  231  Chicago Bulls   
100  2013   Joakim Noah  FC  6-11  230  Chicago Bulls   
122  2012   Joakim Noah  FC  6-11  230  Chicago Bulls   
124  2012     Luol Deng  GF   6-9  220  Chicago Bulls   
128  2011  Derrick Rose  PG   6-3  190  Chicago Bulls   
137  2011     Luol Deng  GF   6-9  220  Chicago Bulls   
170  2010  Derrick Rose  PG   6-3  190  Chicago Bulls   
199  2009  Derrick Rose  PG   6-3  190  Chicago Bulls   

                             Selection Type    NBA Draft Status  \
16      Eastern All-Star Fan Vote Selection  2011 Rnd 1 Pick 30   
29       Eastern All-Star Coaches Selection  2011 Rnd 1 Pick 30   
36   Eastern All-Star Replacement Selection   2001 Rnd 1 Pick 3   
53      Eastern All-Star Fan Vote Selection   2001 Rnd 1 Pick 3   
58       Eastern All-Star Coaches Selection  2011 Rnd 1 Pick 30   
100      Eastern All-Star Coaches Selection   2007 Rnd 1 Pick 9   
122      Eastern All-Star Coaches Selection   2007 Rnd 1 Pick 9   
124      Eastern All-Star Coaches Selection   2004 Rnd 1 Pick 7   
128     Eastern All-Star Fan Vote Selection   2008 Rnd 1 Pick 1   
137      Eastern All-Star Coaches Selection   2004 Rnd 1 Pick 7   
170     Eastern All-Star Fan Vote Selection   2008 Rnd 1 Pick 1   
199      Eastern All-Star Coaches Selection   2008 Rnd 1 Pick 1   

               Nationality               Draft_Class drft_yr  rnd rnd_num  \
16           United States  [2011, Rnd, 1, Pick, 30]    2011  Rnd       1   
29           United States  [2011, Rnd, 1, Pick, 30]    2011  Rnd       1   
36                   Spain   [2001, Rnd, 1, Pick, 3]    2001  Rnd       1   
53                   Spain   [2001, Rnd, 1, Pick, 3]    2001  Rnd       1   
58           United States  [2011, Rnd, 1, Pick, 30]    2011  Rnd       1   
100  France\nUnited States   [2007, Rnd, 1, Pick, 9]    2007  Rnd       1   
122  France\nUnited States   [2007, Rnd, 1, Pick, 9]    2007  Rnd       1   
124   England\nSouth Sudan   [2004, Rnd, 1, Pick, 7]    2004  Rnd       1   
128          United States   [2008, Rnd, 1, Pick, 1]    2008  Rnd       1   
137   England\nSouth Sudan   [2004, Rnd, 1, Pick, 7]    2004  Rnd       1   
170          United States   [2008, Rnd, 1, Pick, 1]    2008  Rnd       1   
199          United States   [2008, Rnd, 1, Pick, 1]    2008  Rnd       1   

      pck pck_num foreign_or_not  
16   Pick      30             US  
29   Pick      30             US  
36   Pick       3             No  
53   Pick       3             No  
58   Pick      30             US  
100  Pick       9             No  
122  Pick       9             No  
124  Pick       7             No  
128  Pick       1             US  
137  Pick       7             No  
170  Pick       1             US  
199  Pick       1             US

SyntaxError: invalid syntax (<ipython-input-126-53ab44f20fbb>, line 1)